### DISCUSSION TITLE
Shared Spinner loader for all livewire components

### DISCUSSION THREAD

**Discussion by abdosaeedelhassan** [livewire/livewire#2355] [1. Help]
Hi, i know that every livewire component has it own loading indicator (wire:loading), my question i want to make shared full page loader for all livewire component that work in loading status and stop after component stop loading, instead of calling it in every component, is there away to do something like this?, thank you

**Comment by artworkad** [comment]
For this you have to use livewire js hooks: https://laravel-livewire.com/docs/2.x/reference

E.g.:

    <script>
        window.onload = function() {
            Livewire.hook('message.sent', () => {
                window.dispatchEvent(
                    new CustomEvent('loading', { detail: { loading: true }})
                );
            })
            Livewire.hook('message.processed', (message, component) => {
                window.dispatchEvent(
                    new CustomEvent('loading', { detail: { loading: false }})
                );
            })
        }
    </script>
    
    <div 
        x-data="{ loading: false }" 
        x-show="loading"
        @loading.window="loading = $event.detail.loading" 
        class="absolute z-50 bottom-20 right-20" 
    >
        This is taking way too long...
    </div>

Place this this code e.g. in your app layout.  Please consider the failed hook, too. You might need to add it to stop loading when a livewire request fails.

**Reply by kresnasatya** [reply]
I think this solution can be alternative for case trigger `wire:loading` in JavaScript. I was open discussion about that case here: https://github.com/livewire/livewire/discussions/5343
